### PR TITLE
Fix the batch normalization layers updates and training loss computation

### PR DESCRIPTION
**Pull request by minseok-ryu**
- fix the batch normalization issue
- fix the training loss computation

### CODE REVIEW

File: src/appfl/algorithm/client_optimizer.py [hunk 1]
```diff
@@ -33,41 +33,64 @@ def update(self):
...
                 loss = self.loss_fn(output, target)
                 loss.backward()
                 optimizer.step()
+                
+                train_loss += loss.item()
+                if output.shape[1] == 1:
+                    pred = torch.round(output)
+                else:
+                    pred = output.argmax(dim=1, keepdim=True)
+                train_correct += pred.eq(target.view_as(pred)).sum().item()
```
Comment (kibaekkim) [inline src/appfl/algorithm/client_optimizer.py:73]: @minseok-ryu This part must be fixing the issue #94. Can you put some comments here about the fix?

**Review by kibaekkim — COMMENTED** [review]
Please see my comment. Otherwise it looks good to me. It could have been better if we could separate the fix with the clean up. Please revise the PR title and description to properly reflect the commits under this PR.

File: src/appfl/algorithm/client_optimizer.py (hunk 1 above)
Comment (minseok-ryu) [reply src/appfl/algorithm/client_optimizer.py:73]: This part is related to the "training loss computation."

File: src/appfl/algorithm/server_federated.py [hunk 2]
```diff
@@ -62,18 +65,26 @@ def update(self, local_states: OrderedDict):
 
         """ residual calculation """
         super(FedServer, self).primal_residual_at_server()
-
+ 
         """ change device """
-        for i in range(self.num_clients):
-            for name, _ in self.model.named_parameters():
+        for i in range(self.num_clients): 
+            for name in self.model.state_dict():
```
Comment (hthieu166) [inline src/appfl/algorithm/server_federated.py:71]: For `BatchNorm` layers, the `state_dict` also keeps track of the `num_batches_tracked` parameters
[What ‘num_batches_tracked’ in the new BN is for?](https://discuss.pytorch.org/t/what-num-batches-tracked-in-the-new-bn-is-for/27097), which is an integer and should not be updated using averaging. Do you think we should handle this param separately? 

Otherwise, this update should work well for most of the parameters.

**Review by hthieu166 — COMMENTED** [review]
This update looks good to me, only one concern about the `num_batches_tracked ` parameter of the `BatchNorm` layer.  Otherwise, we can merge this pull request.

File: src/appfl/algorithm/server_federated.py (hunk 2 above)
Comment (minseok-ryu) [reply src/appfl/algorithm/server_federated.py:71]: 1. From the source code (https://github.com/pytorch/pytorch/blob/10c4b98ade8349d841518d22f19a653a939e260c/torch/nn/modules/batchnorm.py#L58-L81), ``num_batches_tracked`` is not used under the default setting, i.e., ``momentum=0.1``. 

2. If``momentum=None``, then it will use decreasing factor, 1/``num_batches_tracked``, where ``num_batches_tracked`` increases per every iteration. Do you have a reason why we should not use the averaging of ``num_batches_tracked``?
